src/providers/wfs/oapif/qgsoapifprovider.cpp

```diff
@@ -702,27 +714,32 @@ void QgsOapifFeatureDownloaderImpl::run( bool serializeFeatures, long long maxFe
     hasQueryParam = true;
   }
 
-  const QgsRectangle &rect = mShared->currentRect();
+  QgsRectangle rect = mShared->currentRect();
   if ( !rect.isNull() )
   {
-    // Clamp to avoid server errors.
-    const double minx = std::max( -180.0, rect.xMinimum() );
-    const double miny = std::max( -90.0, rect.yMinimum() );
-    const double maxx = std::min( 180.0, rect.xMaximum() );
-    const double maxy = std::min( 90.0, rect.yMaximum() );
-    if ( minx > 180.0 || miny > 90.0 || maxx  < -180.0 || maxy < -90.0 )
+    if ( mShared->mSourceCrs.isGeographic() )
     {
-      // completely out of range. Servers could error out
-      url.clear();
+      // Clamp to avoid server errors.
+      rect.setXMinimum( std::max( -180.0, rect.xMinimum() ) );
+      rect.setYMinimum( std::max( -90.0, rect.yMinimum() ) );
+      rect.setXMaximum( std::min( 180.0, rect.xMaximum() ) );
+      rect.setYMaximum( std::min( 90.0, rect.yMaximum() ) );
+      if ( rect.xMinimum() > 180.0 || rect.yMinimum() > 90.0 || rect.xMaximum()  < -180.0 || rect.yMaximum() < -90.0 )
+      {
+        // completely out of range. Servers could error out
+        url.clear();
+        rect = QgsRectangle();
+      }
     }
-    else if ( minx > -180.0 || miny > -90.0 || maxx < 180.0 || maxy < 90.0 )
+
+    if ( ! rect.isNull() )
     {
       url += ( hasQueryParam ? QStringLiteral( "&" ) : QStringLiteral( "?" ) );
       url += QStringLiteral( "bbox=%1,%2,%3,%4" )
-             .arg( qgsDoubleToString( minx ),
-                   qgsDoubleToString( miny ),
-                   qgsDoubleToString( maxx ),
-                   qgsDoubleToString( maxy ) );
+             .arg( qgsDoubleToString( rect.xMinimum() ),
```

Comment:
if the CRS of the layer is not the default OGC:CRS84 one, then the bbox-crs parameter should be specified (cf https://docs.opengeospatial.org/is/18-058r1/18-058r1.html#_parameter_bbox_crs), and care should be taken to do the proper axis order inversion for CRS with latitude-longitude or northing-easting order. Cf requirement 23.B of https://docs.opengeospatial.org/is/17-069r4/17-069r4.html#_parameter_bbox that mentions in a rather cryptic way that the first value should be "Lower left corner, coordinate axis 1". Slide 9 of https://files.justobjects.nl/presentation/geonovum-2021-api-strategie/OGC-API-Features.pdf makes it more explicit: "bbox-crs: URI of the CRS in which the coordinates are in the bbox parameter; axis order follows the axis order in the CRS"

Reply:
Is there a way to get the CRS in the format with the url? Or could I add a function in `QgsCoordinateReferenceSystem` to return a string in the format like `http://www.opengis.net/def/crs/OGC/1.3/CRS84`? A problem is that the version is not parsed...

About the axis order, from what I could see the currentRect is in the layer CRS, respecting also axis order inverted or not. In that case the clamping would be wrong for crs with inverted axis. I'll check that

Reply:
@domi4484 have a look to QString QgsServerApiUtils::crsToOgcUri

Reply:
If that suits your needs you might consider moving it from server to core.

Reply:
Thanks @elpaso , I moved it to QgsCoordinateReferenceSystem and declared the one in QgsServerApiUtils deprecated.

@rouault bbox-crs paramter in request is implemented in the last commit

Reply:
> bbox-crs paramter in request is implemented in the last commit

axis inversion should also be added if QgsCoordinateReferenceSystem::hasAxisInverted()  ( to be checked how it behaves with OGC:CRS84, where axis order is longitude, latitude, contrary to EPSG:4326 which is latitude, longitude)

Reply:
Ok now it should be good. At the beginning I thought erroneously that the received rectangle was already right inverted.